pkg/assembler/helpers/purl.go

```diff
@@ -37,80 +41,103 @@ func PurlToPkg(purlUri string) (*model.Package, error) {
 	return purlConvert(p)
 }
 
+// purlConvert converts a purl URI into a graphql package node.
 func purlConvert(p purl.PackageURL) (*model.Package, error) {
-	switch p.Type {
+	purlTypeMap := map[string]func(purl.PackageURL) (*model.Package, error){
+		"alpm":             genericHandler,
```

Comment:
retain comments on the status of these purl definitions from the switch case statements.

Reply:
Included the previous comments.